src/Misc/layoutbin/powershell/CapabilityHelpers/CapabilityFunctions.ps1

```diff
@@ -83,6 +83,146 @@ function Add-CapabilityFromRegistry {
     return $true
 }
 
+
+function Add-CapabilityFromRegistryWithLastVersionAvailableForSubkey {
+    <#
+        .SYNOPSIS
+            Retrieves capability from registry for specified key and subkey. Considers that subkey has semver format
+    #>
+    [CmdletBinding()]
+    param(
+        # Prefix name of capability
+        [Parameter(Mandatory = $true)]
+        [string]$PrefixName,
+        # Postfix name of capability
+        [Parameter(Mandatory = $false)]
+        [string]$PostfixName,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('CurrentUser', 'LocalMachine')]
+        [string]$Hive,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Default', 'Registry32', 'Registry64')]
+        [string]$View,
+        # Registry key
+        [Parameter(Mandatory = $true)]
+        [string]$KeyName,
+
+        [Parameter(Mandatory = $true)]
+        [string]$ValueName,
+        # Registry subkey
+        [Parameter(Mandatory = $true)]
+        [string]$Subkey,
+        # Major version of tool to be added as capability
+        [Parameter(Mandatory = $true)]
+        [int]$MajorVersion,
+        # Minimum major version of tool to be added as capability. All versions detected less than this version - will be ignored. 
+        # This is helpful for backward compatibility with already existing logic for previous versions
+        [Parameter(Mandatory = $false)]
+        [int]$MinimumMajorVersion,
+
+        [ref]$Value)
+    try {
+         Write-Host $MajorVersion $MinimumMajorVersion
+        if ($MajorVersion -lt $MinimumMajorVersion) {
+            return $false
+        }
+
+        $wholeKey = Join-Path -Path $KeyName -ChildPath $Subkey
+
+        $capablityValue = Get-RegistryValue -Hive $Hive -View $View -KeyName $wholeKey -ValueName $ValueName
+
+        if (Is-RegistryValueEmpty -Value $capablityValue) {
+            return $false
+        }
```

Comment:
This can be done with `[string]::IsNullOrEmpty($capabilityValue)` without introducing the `Is-RegistryValueEmpty` function.